readthedocs/projects/forms.py

```diff
@@ -542,6 +542,14 @@ def save(self, commit=True):
         self.project.webhook_notifications.add(self.webhook)
         return self.project
 
+    def clean_url(self):
+        url = self.cleaned_data.get('url')
+        if not url:
+            raise forms.ValidationError(
+                _('This field is required.')
```

Comment:
We will probably want to validate that it's a valid URL with a regex or something like that. Maybe we can use Django internals for this by using the proper `URLField`: https://docs.djangoproject.com/en/2.1/ref/forms/fields/#urlfield

This can be done in the `__init__` method and the `clean_url` method may not be needed.

Reply:
The URL field is working fine because the model field is itself defined with models.URLField
https://github.com/rtfd/readthedocs.org/blob/cf071ed66c10beacbb5e46a8832186013d5a1ea9/readthedocs/projects/models.py#L924-L926

It was accepting empty strings because `blank=True` is set. So, we only need to check for empty strings.

Reply:
I guess we want `blank=False` then, not sure how many projects don't have an url

Reply:
@stsewd 
For now, I am pushing the remaining changes.
If any change is done. I will make another commit to this PR.